### PR TITLE
Improve http status all on http error middleware

### PR DESCRIPTION
**Pull request by Jgaldos**
Fixes [#3851](https://github.com/scrapy/scrapy/issues/3851)

on this spider 
```
from scrapy import Spider, Request


class HttpStatusSpider(Spider):
    name = "http_status"

    def start_requests(self):
        http_status = [200, 300, 400, 500]
        urls = [
            f"http://httpbin.org/status/{status}" for status in http_status
        ]
        for url in urls:
            yield Request(
                url=url,
                callback=self.parse,
                meta={
                    "handle_httpstatus_all": False
                }
            )

    def parse(self, response, **kwargs):
        print(f"response url: {response.url}")
```
previous the fix was printing all elements on the list, after the fix just print 200 status

### CODE REVIEW

File: scrapy/spidermiddlewares/httperror.py
Comment (Gallaecio) [inline scrapy/spidermiddlewares/httperror.py:35]: :lipstick: 

```suggestion
        if meta.get('handle_httpstatus_all', False):
```

Comment (Jgaldos) [reply scrapy/spidermiddlewares/httperror.py]: Fixed

**Comment by eLRuLL** [comment]
@Jgaldos could you please improve the documentation too? as @kmike mentioned we could be more descriptive [here](https://docs.scrapy.org/en/latest/topics/spider-middleware.html#std:reqmeta-handle_httpstatus_all)

File: docs/topics/spider-middleware.rst
Comment (Gallaecio) [inline docs/topics/spider-middleware.rst:259]: I think ‘disable it’ here is slightly ambiguous. Maybe ‘disable the effects of the ``handle_httpstatus_list`` key’?

Comment (Jgaldos) [reply docs/topics/spider-middleware.rst:259]: I agree, I am changing it

File: docs/topics/spider-middleware.rst
Comment (Gallaecio) [inline docs/topics/spider-middleware.rst:259]: My bad, sorry!

```suggestion
disable the effects of the ``handle_httpstatus_all`` key.
```

Comment (Jgaldos) [reply docs/topics/spider-middleware.rst:259]: fixed

**Review by Gallaecio — APPROVED** [review]

**Comment by eLRuLL** [comment]
@Gallaecio what should be the procedure to merge this? Are we waiting for another reviewer to accept and merge this? Please let me know if I can help in any way

**Comment by Gallaecio** [comment]
All reviews are more than welcome, but yes, at the moment we require 2 core developer approvals to merge.

**Pull request merged**
